libtenzir/src/tql2/parser.cpp

```diff
@@ -434,19 +434,34 @@ class parser {
     // TODO: Drop one of the syntax possibilities.
     if (peek(tk::meta) || peek(tk::at)) {
       auto begin = location{};
+      auto is_at = false;
       if (auto meta = accept(tk::meta)) {
         begin = meta.location;
         expect(tk::dot);
       } else {
         begin = expect(tk::at).location;
+        is_at = true;
       }
       auto ident = expect(tk::identifier).as_identifier();
       auto kind = static_cast<enum meta_extractor::kind>(0);
       if (ident.name == "tag") {
         kind = meta_extractor::schema;
+      } else if (ident.name == "import_time") {
+        kind = meta_extractor::import_time;
+      } else if (ident.name == "internal") {
+        kind = meta_extractor::internal;
+      } else if (ident.name == "schema") {
+        diagnostic::error("use `{}tag` instead", is_at ? "@" : "meta.")
```

Comment:
I like `@` a lot as symbol. How is `meta.` not ambiguous with actual values in the data?

Reply:
I implemented both for now, so that we can play around with it and see what's better.

It's not ambiguous in the same way that `this`/`self`/`if`/`match`/any other keyword is not ambiguous: You are just not allowed to use that syntax if your field name is a keyword. You'd use `this["if"]` as a fallback.

Reply:
> You are just not allowed to use that syntax

So I'd get an error if I use `meta.timestamp` and a field with name `meta` exists also?